Improve performance of finding files

crazytonyli opened the pull request.

Dschee commented:
@crazytonyli Thank you for your PR!

I'm looking forward to merging this, but two notes:

1. Please never include anything else into a PR than what it's about. Specifically I'm referring to the Xcode 9.3 SwiftLint warnings that you fixed here. Please always create a seperate PR for that. I have now already updated the code in the stable branch, so rebasing your branch onto the updated stable should be the way to go now.

2. Could you please add performance tests to the project that prove that your approach actually is faster? Feel free to add many files into the Tests/Assets folder to have a bigger amount of files to be traversed.

crazytonyli commented:
@Dschee Thanks. I made those swiftlint changes just to make the code compile, it might be different swiftlint setup on our machines if they are just warnings on yours, but I have rebased and removed those commits, so should be all good now. I've also added performance test and several dummy directories into current test assets.

Dschee commented:
Looks good, merging.
